src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,106 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovies, setNewMovies] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+
+    setNewMovies(prevMovie => ({ ...prevMovie, [name]: value }));
+  };
+
+  const newMovieData =
+    newMovies.title.trim() &&
+    newMovies.imgUrl.trim() &&
+    newMovies.imdbUrl.trim() &&
+    newMovies.imdbId.trim();
+
+  function addMovie() {
+    onAdd(newMovies);
+  }
+
+  function clearForm() {
+    setNewMovies({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
```

Comment:
reuse the variable we create for out state
```suggestion
    setNewMovies(initialMovieState);
```